dipy/reconst/dki.py

```diff
@@ -0,0 +1,1606 @@
+    DT_elements = result[:6]
+    evals, evecs = decompose_tensor(from_lower_triangular(DT_elements),
+                                    min_diffusivity=min_diffusivity)
+
+    # Extracting kurtosis tensor parameters from solution
+    MD_square = (evals.mean(0))**2  
+    KT_elements = result[6:21] / MD_square
+
+    # Write output  
+    dki_params = np.concatenate((evals, evecs[0], evecs[1], evecs[2], 
+                                 KT_elements), axis=0)
+
+    return dki_params
+
+
+def wls_fit_dki(design_matrix, data, min_signal=1):
+    r"""
+    Computes weighted linear least squares (WLS) fit to calculate
+    the diffusion tensor and kurtosis tensor using a weighted linear 
+    regression diffusion kurtosis model [1]_.
+
+    Parameters
+    ----------
+    design_matrix : array (g, 22)
+        Design matrix holding the covariants used to solve for the regression
+        coefficients.
+    data : array (..., g)
+        Data or response variables holding the data. Note that the last
+        dimension should contain the data. It makes no copies of data.
+    min_signal : default = 1
+        All values below min_signal are repalced with min_signal. This is done
+        in order to avoid taking log(0) durring the tensor fitting.
+
+    Returns
+    -------
+    dki_params : array (..., 27)
+        All parameters estimated from the diffusion kurtosis model for all N
+        voxels. 
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor 
+
+    See Also
+    --------
+    decompose_tensors
+
+
+    References
+    ----------
+       [1] Veraart, J., Sijbers, J., Sunaert, S., Leemans, A., Jeurissen, B.,
+           2013. Weighted linear least squares estimation of diffusion MRI
+           parameters: Strengths, limitations, and pitfalls. Magn Reson Med 81,
+           335-346.
+    """
+
+    tol = 1e-6
+    if min_signal <= 0:
+        raise ValueError('min_signal must be > 0')
+
+    # preparing data and initializing parametres
+    data = np.asarray(data)
+    data_flat = data.reshape((-1, data.shape[-1]))
+    dki_params = np.empty((len(data_flat), 27))
+
+    # inverting design matrix and defining minimun diffusion aloud
+    min_diffusivity = tol / -design_matrix.min()
+    inv_design = np.linalg.pinv(design_matrix)
+
+    # lopping WLS solution on all data voxels
+    for vox in range(len(data_flat)):
+        dki_params[vox] = _wls_iter(design_matrix, inv_design, data_flat[vox],
+                                    min_signal, min_diffusivity)
+
+    # Reshape data according to the input data shape
+    dki_params = dki_params.reshape((data.shape[:-1]) + (27,))
+
+    return dki_params
+
+
+def _wls_iter(design_matrix, inv_design, sig, min_signal, min_diffusivity):
```

Comment:
The function dipy.recont.dti is implemented specifically for the DTI, so we cannot call this function here. This will be only possible if we generalize the DTI fitting algorithms. For this, we have to remove the decompose_tensor step and change the variable output format. Tensor decomposition have to be done instead in the main. This changes are feasible, however I will recommend doing this in another pull request, this one is getting long.